src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php

```diff
@@ -111,7 +111,7 @@ public function generateForPath(StructureShape $shape, string $path, string $out
 
     protected function parseResponseTimestamp(Shape $shape, string $input, bool $required): string
     {
-        $body = 'new \DateTimeImmutable((string) INPUT)';
+        $body = 'is_numeric(INPUT) ? \Datetime::createFromFormat("U.u", sprintf("%.6F", INPUT)) : new \DateTimeImmutable((string) INPUT)';
```

Comment:
What is the prupose of the check on `is_numeric` do you have example of non numeric data?

Reply:
Well I did not want to break anything on other Client as I don't know how other timestamp are returned by aws !

But if we are sure we have real timestamp everywhere I can remove it no problem !

Reply:
I think the `sprintf` thing would fix the input. https://3v4l.org/lXAQc#v7.2.34

Reply:
hmmm... looks like this is already what we are doing in `JsonRpcParser`